### PR TITLE
add automatic relay deployment logic

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
Automatic relay deployment.

### CODE REVIEW

File: eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt [hunk 1]
```diff
@@ -69,6 +69,29 @@ class RelayRegistration(
...
+                logger.info { "Relay replenishment triggered" }
+
+                freeRelayProvider.getRelays().map { relays ->
+                    logger.info { "Free relays: ${relays.size}" }
+                    val toDeploy = relayRegistrationConfig.number - relays.size
+                    if (toDeploy > 0)
+                        deploy(toDeploy, relayRegistrationConfig.ethMasterWallet)
+                }.failure { throw it }
+
+                runBlocking { delay(relayRegistrationConfig.replenishmentPeriod * 1000) }
```
Comment (dolgopolovwork) [inline eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt]: Why can't we use `Thread.sleep(relayRegistrationConfig.replenishmentPeriod * 1000) `instead?

File: eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt [hunk 2]
```diff
@@ -69,6 +69,29 @@ class RelayRegistration(
...
 
+    /**
+     * Run a job that every replenishmentPeriod checks that number from config free relays are present. In case of
+     * lack of free relays deploys lacking amount.
+     */
+    fun runRelayReplenishment(): Result<Unit, Exception> {
+        logger.info { "Run relay replenishment" }
+
+        return Result.of {
+            while (true) {
```
Comment (dolgopolovwork) [inline eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt]: I think we can make it easier. Please, take a look at [ScheduledExecutorService](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html)

Comment (Mingela) [reply eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt]: I think it is not used for purpose since it is about async execution

Comment (dolgopolovwork) [reply eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt]: Do you mean `IrohaNetworkImpl`? I thought it's a thread-safe class

File: eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt]: https://medium.com/@mohakpuri1712/kotlin-coroutines-thread-sleep-vs-delay-63171fe8a24

Comment (Alexey-N-Chernyshov) [reply eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt]: There is explained a difference between Thread.sleep() and delay(). delay() is non-blocking suspend function. However, there is no need to use it.

**Pull request merged**